features/cellular/framework/targets/TELIT/ME910/TELIT_ME910.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2017, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CELLULAR_TARGETS_TELIT_ME910_TELIT_ME910_H_
+#define CELLULAR_TARGETS_TELIT_ME910_TELIT_ME910_H_
+
+#ifdef TARGET_FF_ARDUINO
+#ifndef MBED_CONF_TELIT_ME910_TX
+#define MBED_CONF_TELIT_ME910_TX D1
+#endif
+#ifndef MBED_CONF_TELIT_ME910_RX
+#define MBED_CONF_TELIT_ME910_RX D0
+#endif
+#endif /* TARGET_FF_ARDUINO */
+
+#include "AT_CellularDevice.h"
+
+//the delay between sending AT commands
+#define DEFAULT_DELAY_BETWEEN_AT_COMMANDS 20
+
+namespace mbed {
+
+class TELIT_ME910 : public AT_CellularDevice {
+public:
+    TELIT_ME910(FileHandle *fh);
+
+protected: // AT_CellularDevice
+    virtual uint16_t get_send_delay() const;
+    virtual nsapi_error_t init();
+    virtual nsapi_error_t hard_power_on();
```

Comment:
Need to be implemented if defined, or if don't have the default power implementation the declaration of power-functions should be removed.

Reply:
@AriParkkila I added the default power implementation for the ME910. I was toying around with this since we have a separate, board-specific power enable pin that I had to accommodate, but that functionality doesn't make sense here (should be in our application/BSP code), so now I just have the necessary power on sequences as suggested in the ME910 manual.

Reply:
@AriParkkila I see that a couple of the CI jobs are failing because my power implementation references the `MDMPWRON` pin:

```
[Error] TELIT_ME910.cpp@95,29: 'MDMPWRON' was not declared in this scope
[Error] TELIT_ME910.cpp@109,29: 'MDMPWRON' was not declared in this scope
```

Is there a recommended way to handle this? Use a #if check for the `MDMPWRON` pin? Use a #if check for `MODEM_ON_BOARD`?

Is it better to just use the default power implementation here in the driver and leave it up to the end user to implement this in an `onboard_modem_api.c` file?

Reply:
@trowbridgec can you follow the implementation from [EC2X](https://github.com/ARMmbed/mbed-os/tree/master/features/cellular/framework/targets/QUECTEL/EC2X).

Reply:
@trowbridgec You should declare the pins as class variables, e.g. `TELIT_ME910.h`:
```
    bool _active_high;
    DigitalOut _pwr;
    DigitalOut _rst;
```
And then use the variables instead of compile-time defines, e.g. `TELIT_ME910.cpp`:
```
nsapi_error_t TELIT_ME910::hard_power_on()
{
    if (_pwr.is_connected()) {
        _pwr = _active_high;
```
In addition, the pins need to be provided in the class constructor.